R/derive_vars_aage.R

```diff
@@ -132,6 +132,13 @@ derive_var_age_years <- function(dataset, age_var, age_unit = NULL, new_var) {
   age_var <- age_variable
   unit_var <- paste0(quo_get_expr(age_var), "U")
 
+  if (!is.null(age_unit)) {
+    if (length(age_unit) > 1) {
+      err_msg <- "Only one unit can be specified via the `age_unit` parameter"
+      abort(err_msg)
+    }
+  }
+
```

Comment:
Can we use any of our `assert` functions here?  https://pharmaverse.github.io/admiraldev/devel/reference/index.html

Reply:
Thanks for your suggestion. I've moved the call to **assert_character_scalar()**, previously at line 146, to nearer the start of the function, and I used that to replace the code that I wrote to check **age_unit**. I also amended the call to **assert_character_scalar()**, assigning values to the parameters **case_sensitive** and **optional**.
There is possibly a bug with **assert_character_scalar()**, which I've raised as issue number 156 in {admiraldev}, though I'm no longer able to reproduce the issue.